packages/mobile/src/store/offline-downloads/sagas/downloadQueueSagas/syncCollectionWorker.ts

```diff
@@ -0,0 +1,222 @@
+import type { DownloadReason, ID } from '@audius/common'
+import {
+  reachabilityActions,
+  cacheCollectionsSelectors,
+  accountSelectors,
+  getContext
+} from '@audius/common'
+import { difference } from 'lodash'
+import moment from 'moment'
+import { call, put, race, select, take } from 'typed-redux-saga'
+
+import { DOWNLOAD_REASON_FAVORITES } from 'app/services/offline-downloader'
+import { dispatch } from 'app/store/store'
+
+import {
+  getCollectionSyncStatus,
+  getOfflineCollectionMetadata,
+  getOfflineTrackMetadata
+} from '../../selectors'
+import type { CollectionId, OfflineItem } from '../../slice'
+import {
+  completeCollectionSync,
+  CollectionSyncStatus,
+  errorCollectionSync,
+  cancelCollectionSync,
+  requestDownloadQueuedItem,
+  redownloadOfflineItems,
+  addOfflineItems,
+  removeOfflineItems,
+  startCollectionSync
+} from '../../slice'
+
+const { SET_UNREACHABLE } = reachabilityActions
+const { getUserId } = accountSelectors
+const { getCollection } = cacheCollectionsSelectors
+
+const isTrackFavoriteReason = (downloadReason: DownloadReason) =>
+  downloadReason.is_from_favorites &&
+  downloadReason.collection_id === DOWNLOAD_REASON_FAVORITES
```

Comment:
should this be `||`? I may be wrong and would defer to @amendelsohn but i thought that favorites would not have `is_from_favorites` set to true so this would never return true?

Reply:
a favorited track will be `{ is_from_favorites: true, collection_id: 'favorites' }` so this is correct

Reply:
yes i see below now. thanks 👍